schemas/reports/document_with_address_information_report.yaml

```diff
@@ -1,5 +1,10 @@
 allOf:
   - $ref: report_common.yaml
+  - type: object
+    properties:
+      name:
+        type: string
+        enum: [document_with_address_information, no_value]
```

Comment:
What's `no_value` in here? I can't find any reference in the OpenAPI docs to this

Reply:
Yeah, actually it's just a temporary remediation for a bug in openapi converter (at least python one).

First thing, let ignore `, no_value` for a while. Setting enum this way it's to enforce a constant value in openapi standard. That allow to select the good report depending on the value of this field.

Problem is that gets converted, at least in python, into:
```python
        if value not in ('document_with_address_information'):
            raise ValueError("must be one of enum values ('document_with_address_information')")
```
That's a problem as it become a substring match, i.e. `document` is valid. Adding `no_value` is an hack to get code below generated:
```python
        if value not in ('document_with_address_information', 'no_value'):
            raise ValueError("must be one of enum values ('document_with_address_information', 'no_value')")
```
Checkng if value is included into a tuple of two values, which is working right.
This is just a remediation to gain some time to have generator fixed.

Reply:
Ah got it!

Reply:
is setting this "enum" required? the "report.yaml" spec already descriminates based on "name".

Reply:
Need to double check other converters but `discriminator` approach we were using earlier is not working at least in python, probably due to the fact we're using `anyOf` and `allOf` at different levels:
```python
discriminator_value_class_map: Dict[str, str] = {
    }
```

Reply:
can you share the python snippet/test which fails because of this?

Reply:
Yeah, sure. Here you go:
```python
with onfido.ApiClient(configuration) as api_client:
  onfido_api = onfido.DefaultApi(api_client)
  check = onfido_api.create_check(onfido.Check(type='express', applicant_id=applicant.id,
                                               report_names=['identity_enhanced', 'document', 'facial_similarity_photo']))

  reports_list = onfido_api.list_reports(check_id=check.id)
```
Which raises:
```
ValueError: Multiple matches found when deserializing the JSON string into Report with oneOf schemas: DocumentReport, DocumentWithAddressInformationReport, DocumentWithDriverVerificationReport, DocumentWithDrivingLicenceInformationReport, FacialSimilarityPhotoReport, FacialSimilarityVideoReport, IdentityEnhancedReport, KnownFacesReport, PhotoFullyAutoReport, ProofOfAddressReport, RightToWorkReport, WatchlistEnhancedReport, WatchlistPepsOnlyReport, WatchlistSanctionsOnlyReport, WatchlistStandardReport.
```

Reply:
It seems we need to enable the `useOneOfDiscriminatorLookup` [option](https://openapi-generator.tech/docs/generators/python/#:~:text=useOneOfDiscriminatorLookup) in python generator to have that discriminator logic working. New commit coming soon.